### DISCUSSION TITLE
How do you want to learn about VisiData features?

### DISCUSSION THREAD

**Discussion by anjakefala** [saulpw/visidata#2221] [Q&A]
We are looking to invest more effort into VisiData learnability.

We wanted to know:

    What are the ways you learned how to use VisiData?
    How would you like to learn more about VisiData in the future?

We can provide more information on the direction we are planning to go in, and ask for your feedback on it, but I wanted to hear your un-biased impression first!

**Comment by halloleo** [comment]
I learn most about Visidata through the documentation. Sometimes I search the man page. However, discoverability of commands feels to me  pretty difficult, but that's maybe just me being a very casual user

Also, the mix of documentation, tutorials and manpage is sometimes confusing. Don't get me wrong: It needs all three for sure, but I often lost for where to find what I need...

**Reply by anjakefala** [reply]
I really appreciate your reply!

> I learn most about Visidata through the documentation.

By "documentation" do you mean https://visidata.org/docs?

**Reply by halloleo** [reply]
Yes, I that's what I mean.

**Reply by halloleo** [reply]
Also, an example of my confusion between the different "channels" (docs, tuts & manpage) is actually what lead me to my [latest question](https://github.com/saulpw/visidata/discussions/2222): I looked at the ["How to" about Columns](https://www.visidata.org/docs/columns/) and was hoping to find something about search-&-replace in a column, but could find anything there, but maybe I just didn't look properly..

**Comment by maxim-uvarov** [comment]
> What are the ways you learned how to use VisiData?

I've learned what I know (and it is only basics) using: 
1. https://jsvine.github.io/intro-to-visidata/index.html
2. https://visidata.org/docs
3. Reading github discussions.

> How would you like to learn more about VisiData in the future?

As for me, the most desirable way would be a tutorial like `vimtutor` or the one from the Helix editor `hx --tutor`. Additionally, videos with examples of VisiData usage would be helpful.

**Reply by jgarte** [reply]
`vdtutor` would be an awesome feature for learning VisiData!

**Comment by reagle** [comment]
Same as @maxim-uvarov. I would've *liked* to learn via sidebar (now available!), command palette (now available!), Anki deck (beta quality), FAQ, and chatbot.

**Comment by gennaro-tedesco** [comment]
I would daresay that `visidata` may gain a much wider audience if a relevant YouTube (or similar platform) channel were to be updated: I know that the author already invested time in some didactic videos some time ago, but they are probably out-of-date with the recent developments; generally a series of short&sweet videos showcases on how to use the software (from trivial to more elaborate use cases) would go a long way!

**Reply by saulpw** [reply]
Thanks for weighing in, @gennaro-tedesco!  One of the problems with videos is exactly as you say--they go out-of-date and can't be updated.  Also I have a hard time with doing all of the steps myself--writing the script, being on camera, editing the video, and adding the metadata to the result.  But I'd love to work with someone who was willing to do some of these steps!

**Reply by reagle** [reply]
I'm naive, I'm sure, but I think I'd enjoy unscripted and unedited screen recordings of people thinking through challenges, giving voice to what they are doing and seeing the key strokes on screen. [Hanukkah of Data](https://www.visidata.org/blog/2022/hanukkah-of-data/) comes to mind, but I'm not sure if there are others. (The 1B row challenge is more about performance.) Last week I somehow encountered [AndyMath](https://www.youtube.com/@AndyMath) challenges, and even though I haven't thought about geometry and algebra in decades, they are awesome! Obviously, a different type of thing, but I wonder if there's something similar for data analysis.

**Reply by saulpw** [reply]
@reagle You probably would enjoy unscripted/unedited recordings from someone who is not me :)  If it's not scripted, I get caught in brain loops and just stare off into space for 15-30 seconds while I try to get back on track.  I think we've all been spoiled by Youtube creators who make it seem effortless to produce an offhand video, but in reality they put a large amount of effort that effortlessness.

**Reply by reagle** [reply]
As an aside, this is kind of task in video editing is one of the areas where I hope [AI proves to be useful](https://www.veed.io/magic-cut).

**Comment by Mickychen00** [comment]
Will there be a CDN speed-up for readers from different countries?

**Reply by saulpw** [reply]
Hi @Mickychen00, I'm not sure I understand the question, can you explain?

**Reply by Mickychen00** [reply]
Hi. I found that the documentation website is a little slow to respond to click actions. In other words, I think the website needs a CDN speed-up for readers from different IPs.

**Reply by saulpw** [reply]
visidata.org is a static site served via netlify, which should already be serving the content from different servers for different locations.  Sorry, but if your experience of the site is slow, there's not much we can do about it.

**Comment by daviewales** [comment]
1. Tutorial: https://jsvine.github.io/intro-to-visidata/index.html
2. Docs: https://visidata.org/docs
3. Man page: (`gCtrl+H`)
4. Command list: (`zCtrl+H`)
5. Menus
6. GitHub Discussions: https://github.com/saulpw/visidata/discussions
7. GitHub Issues: https://github.com/saulpw/visidata/issues
8. Saul's Visidata playlist on YouTube: https://www.youtube.com/playlist?list=PLxu7QdBkC7drrAGfYzatPGVHIpv4Et46W
9. API Docs: https://www.visidata.org/docs/api/
10. Source code: https://github.com/saulpw/visidata

**Reply by halloleo** [reply]
Does the rank of a learning resource in the list indicate the importance of the resource?

**Reply by daviewales** [reply]
Originally the list was in order of beginner to advanced resources, but some items are just in order of when I remembered them while typing...

**Comment by Zeethulhu** [comment]
VisiData is such a versatile tool and can be used is many different applications.

I often find myself in a situation where I know VisiData **can** get me the solution I want, but I'm not sure **how** to get there. Sometimes getting the solution takes longer than I feel it should.

I would love to see a Cookbook of uses from different views. Drawing inspiration from different approaches and seeing the practical uses for many of Visidata's functionality would help illustrate their use and improve understanding.

I work in Cloud Infrastructure/DevOps and I use VisiData for viewing and searching logs, converting files, exploring AWS resources. But I might learn some new techniques by looking at other uses. Data munging, performing finance calculations, web scraping, etc.

To make submissions easier and encourage contributions, perhaps instructions and a template. Submissions include sample data file, command log file (.vdj), couple of paragraphs explaining the goal and some of the steps. With the Command Log / Replay file it would be easy to rerun the steps for a video and include that, or the reader can download it and see the interactive replay on their computer with the sample data. (Feels easier than writing something up and including all the keystrokes)

As an aside, I feel it would be nice to have option for an alternate replay file based on YAML, for easier reading and support for comments, but that might just be my DevOps tendencies speaking. (Worthy of raising an Idea 💡?)

**Reply by anjakefala** [reply]
We would love to welcome contributions to our guides: https://github.com/saulpw/visidata/issues/2313

These represent our current documentation effort. I really encourage you to try contributing to it. =)

**Reply by anjakefala** [reply]
> As an aside, I feel it would be nice to have option for an alternate replay file based on YAML, for easier reading and support for comments, but that might just be my DevOps tendencies speaking. (Worthy of raising an Idea 💡?)

Try saving a cmdlog as [.vdx](https://github.com/saulpw/visidata/commit/cb8c72a76f1057576012daf2d6984717fa0fc0e2). I'm curious how you'll find it. =) It is an experimental format.
